detection_rules/rule.py

```diff
@@ -367,6 +369,9 @@ def get_required_fields(self, index: str) -> List[dict]:
             if beat_schema and not is_ecs:
                 field_type = beat_schema.get(fld, {}).get('type')
 
+            if endgame_schema:
+                field_type = endgame_schema.endgame_schema.get(fld, None)
```

Comment:
type error here - `endgame_schema` is a dict

Reply:
looks like the problem is with type hinting from the method

Reply:
This won't work as is - if a rule has beats-specific fields and endgame in the index, then endgame will always overwrite.

Reply:
Shouldn't they be the exact same?

Reply:
Shouldn't the be the exact same since they're both ECS?

Reply:
Do we even need to do this here in `get_required_fields`, since we validate in `rule_validators`?

Reply:
Modified to only check if the field is_ecs